src/EventSubscriber/CacheableResponseSubscriber.php

```diff
@@ -21,14 +22,42 @@ class CacheableResponseSubscriber implements EventSubscriberInterface {
    */
   protected $logger;
 
+  /**
+   * Configuration Factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  protected $configFactory;
+
   /**
    * Constructs a new DefaultExceptionHtmlSubscriber.
    *
    * @param \Psr\Log\LoggerInterface $logger
    *   The logger service.
+   * @param \Drupal\Core\Config\ConfigFactory $configFactory
+   *   Configuration for this module.
    */
-  public function __construct(LoggerInterface $logger) {
+  public function __construct(LoggerInterface $logger, ConfigFactory $configFactory) {
```

Comment:
This is the standard way of handling new constructor arguments:

```suggestion
  public function __construct(LoggerInterface $logger, ConfigFactoryInterface $config_factory = NULL) {
    if (!$config_factory instanceof ConfigFactoryInterface) {
      @trigger_error('Not passing the config factory service as the second parameter to ' . __METHOD__ . ' is deprecated in pantheon_advanced_page_cache:1.1.0 and will throw a type error in pantheon_advanced_page_cache:2.0.0. Pass an instance of \\Drupal\\Core\\Config\\ConfigFactoryInterface. See https://www.drupal.org/node/2944229', E_USER_DEPRECATED);
      $config_factory = \Drupal::service('config.factory');
    }
```